emsdk.py

```diff
@@ -2628,15 +2597,14 @@ def construct_env(tools_to_activate, permanent):
   em_config_path = os.path.normpath(dot_emscripten_path())
   if to_unix_path(os.environ.get('EM_CONFIG', '')) != to_unix_path(em_config_path):
     env_vars_to_add += [('EM_CONFIG', em_config_path)]
-  if emscripten_config_directory == emsdk_path():
-    # Remove this once emscripten support CACHE in the config file:
-    # https://github.com/emscripten-core/emscripten/pull/11091
-    em_cache_dir = sdk_path('.emscripten_cache')
-    if to_unix_path(os.environ.get('EM_CACHE', '')) != to_unix_path(em_cache_dir):
-      env_vars_to_add += [('EM_CACHE', em_cache_dir)]
-    mkdir_p(em_cache_dir)
 
   for tool in tools_to_activate:
+    config = tool.activated_config()
+    if 'EMSCRIPTEN_ROOT' in config:
+      # For older emscripten versions that don't use this default we export
+      # EM_CACHE.
```

Comment:
don't all configs have `EMSCRIPTEN_ROOT`?

but regardless I'm confused here. if that's in the config, does that mean this config "doesn't use this default" - ? I'm not sure how to understand that phrase and connect it to this code.

Reply:
Regarding the first point.  No EMSCRIPTEN_ROOT doesn't always exists. For example if you just install a single tool:

```
./emsdk install node-12.9.1-64bit,
./emsdk activate node-12.9.1-64bit,
./emsdk_env.sh
```

To answer the second point:

New emscripten: Uses in-tree cache by default
Old emscripten: Needs to be told to use in-tree case via EM_CACHE environment variable.

In neither case is that CACHE or EM_CACHE part of the activated_config.

Reply:
I see, thanks!